fmi/server_http.go

```diff
@@ -0,0 +1,130 @@
+package fmi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type ServerHTTP struct {
+	address         string
+	mux             *http.ServeMux
+	shutdownTimeout time.Duration
+	taskRunner      TaskRunner
+}
+
+// NewServerHTTP creates a new ServerHTTP.
+func NewServerHTTP(config ServerConfig, insecureBMC bool) (*ServerHTTP, error) {
+	mux := http.NewServeMux()
+	taskRunner, err := NewTaskRunner(config.TaskRunnerType, config.KubeClient, insecureBMC)
+	if err != nil {
+		return nil, err
+	}
+	server := &ServerHTTP{
+		address:         fmt.Sprintf("%s:%d", config.Hostname, config.Port),
+		mux:             mux,
+		shutdownTimeout: config.ShutdownTimeout,
+		taskRunner:      taskRunner,
+	}
+	server.registerRoutes()
+	return server, nil
+}
+
+// Start starts the server.
+func (s *ServerHTTP) Start(ctx context.Context) error {
+	g, ctx := errgroup.WithContext(ctx)
+	server := &http.Server{
+		Addr:    s.address,
+		Handler: s.mux,
+	}
+
+	g.Go(func() error {
+		<-ctx.Done()
+		shutdownCtx, cancel := context.WithTimeout(context.Background(), s.shutdownTimeout)
+		defer cancel()
+		return server.Shutdown(shutdownCtx)
+	})
+
+	g.Go(func() error {
+		return server.ListenAndServe()
+	})
+
+	return g.Wait()
+}
+
+// registerRoutes registers the server's routes.
+func (s *ServerHTTP) registerRoutes() {
+	s.mux.HandleFunc("/scan", s.scanHandler)
+	s.mux.HandleFunc("/settings-apply", s.settingsApplyHandler)
+	s.mux.HandleFunc("/version-update", s.versionUpdateHandler)
```

Comment:
idea: would be great to have an endpoint to get all current active tasks from the sync.Map

Reply:
@stefanhipfel thanks for idea.
Totally agree. Also we might need to implement queues for requests, rate limiting, retrying on client side and tons of other stuff. Apart from that, I'd also like to generate http API from spec without hardcoding endpoints.

But for now, I think the main goal is to come to agreement - will we proceed further with this design or not.